### PR TITLE
Add support for the "neuron-id-to-candid-subaccount" command

### PR DESCRIPTION
**Pull request by anchpop**
Pretty much all the implementation is already done in the ic repo, we just need to hook it up to the dfx extension

### CODE REVIEW

File: extensions/sns/extension.json [hunk 1]
```diff
@@ -77,6 +77,17 @@
         }
       }
     },
+    "neuron-id-to-candid-subaccount": {
+      "about": "Converts a Neuron ID to a candid subaccount blob suitable for use in the `manage_neuron` method on SNS Governance",
+      "args": {
+        "neuron_id": {
+          "about": "The (hex-encoded) SNS neuron ID to convert to a candid subaccount blob"
+        },
```
Comment (anchpop) [inline extensions/sns/extension.json:85]: This is a positional argument, I wasn't sure how to denote that

Comment (ericswanson-dfinity) [reply extensions/sns/extension.json:85]: Not having `long` or `short` should indicate that this is positional (so no change needed)

File: extensions/sns/extension.json [hunk 2]
```diff
@@ -77,6 +77,17 @@
...
+        "escaped": {
+          "about": "If true, print the escaped version of the candid string, useful for pasting into a bash script for example. Default is false"
```
Comment (ericswanson-dfinity) [inline extensions/sns/extension.json:87]:
```suggestion
          "about": "If true, print the escaped version of the candid string, useful for pasting into a bash script for example. Default is false",
          "long": "escaped"
```

File: extensions/sns/e2e/tests/sns.bash [hunk 3]
```diff
@@ -206,3 +206,14 @@ SNS_CONFIG_FILE_NAME="sns_init.yaml"
...
+# This test asserts that the `neuron-id-to-candid-subaccount` subcommand exist in the current extension version.
+@test "sns neuron-id-to-candid-subaccount exists" {
+    run dfx sns neuron-id-to-candid-subaccount --help
+    assert_output --partial "Converts a Neuron ID to a candid subaccount blob"
+}
+# check the output of a particular case of neuron-id-to-candid-subaccount
+@test "sns neuron-id-to-candid-subaccount has a reasonable output" {
+    run dfx sns neuron-id-to-candid-subaccount 9f5f9fda77a03e7177126d0be8c99e931a5381731d00da53ede363140e1be5d6
+    assert_output "blob \"\\9f\\5f\\9f\\da\\77\\a0\\3e\\71\\77\\12\\6d\\0b\\e8\\c9\\9e\\93\\1a\\53\\81\\73\\1d\\00\\da\\53\\ed\\e3\\63\\14\\0e\\1b\\e5\\d6\""
+}
```
Comment (ericswanson-dfinity) [inline extensions/sns/e2e/tests/sns.bash:219]: can we have a test that shows what `--escaped` does?

Comment (anchpop) [reply extensions/sns/e2e/tests/sns.bash:223]: Added!

Comment (anchpop) [reply extensions/sns/e2e/tests/sns.bash:223]: Added!

**Review by ericswanson-dfinity — APPROVED** [review]

**Pull request merged**